department-of-veterans-affairs/caseflow-efolder
# AlecK/APPEALS-26715 Setup monitoring for deprecation warnings

APPEALS-26715 Setup monitoring for deprecation warnings in production

### Description
Added over monitoring and deprecation warnings for Slack and Sentry

### Acceptance Criteria
Check to see if we are already capturing deprecation warnings in production (in `Sentry`)
If we are not, set up a Rails initializer to send deprecation warnings to a monitoring tool (`Sentry`)
[See Fast Ruby: The Complete Guide for Deprecation Warnings in Rails](https://www.fastruby.io/blog/rails/upgrades/deprecation-warnings-rails-guide.html): Finding for inspiration.
Wire up alerts for deprecation warnings to an appropriate Slack channel (we may wish to create a dedicated channel)

### Testing Plan
- Within the rails console:
    - Run ActiveSupport::Deprecation.warn("test deprecation warning")
    - Check Sentry and #appeals-deprecation-alerts Slack channel to ensure they received messages
